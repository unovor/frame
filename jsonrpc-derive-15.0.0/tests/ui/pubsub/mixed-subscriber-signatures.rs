#[macro_use]
extern crate jsonrpc_derive;
extern crate jsonrpc_core;
extern crate jsonrpc_pubsub;

#[rpc]
pub trait Rpc {
	type Metadata;

	/// Hello subscription
	#[pubsub(subscription = "hello", subscribe, name = "hello_subscribe", alias("hello_sub"))]
	fn subscribe(&self, _: Self::Metadata, _: typed::Subscriber<String>, _: u64);

	/// Hello subscription with probably different impl and mismatched Subscriber type
	#[pubsub(subscription = "hello", subscribe, name = "hello_anotherSubscribe")]
	fn another_subscribe(&self, _: Self::Metadata, _: typed::Subscriber<usize>, _: u64);

	/// Unsubscribe from hello subscription.
	#[pubsub(subscription = "hello", unsubscribe, name = "hello_unsubscribe")]
	fn unsubscribe(&self, _: Option<Self::Metadata>, _: SubscriptionId) -> Result<bool>;
	// note that the unsubscribe method is missing
}

fn main() {}
