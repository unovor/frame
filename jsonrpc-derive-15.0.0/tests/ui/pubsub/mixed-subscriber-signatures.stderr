error: Inconsistent signature for 'Subscriber' argument: typed :: Subscriber < usize >, previously defined: typed :: Subscriber < String >
  --> $DIR/mixed-subscriber-signatures.rs:16:52
   |
16 |     fn another_subscribe(&self, _: Self::Metadata, _: typed::Subscriber<usize>, _: u64);
   |                                                       ^^^^^^^^^^^^^^^^^^^^^^^^
